src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+import core.basesyntax.figure.*;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier supplier = new ColorSupplier();
+    public Figure getRandomFigure() {
+        Color color = supplier.getRandomColor();
+        return switch (random.nextInt(5)) {
+            case 0 -> getRandomCircle(color);
+            case 1 -> getRandomRectangle(color);
+            case 2 -> getRandomIsoscelentTrapezoid(color);
+            case 3 -> getRandomSquare(color);
+            case 4 -> getRandomRightTriangle(color);
+            default -> null;
```

Comment:
Returning `null` from a method is a bad practice. Instead of returning `null`, consider handling the default case more appropriately, such as throwing an exception or returning a default figure.